1-mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,5 +1,38 @@
-function setAlarm() {}
+function setAlarm() {
+  const timeRemaining = document.querySelector("#timeRemaining");
+  let setAlarmInput = document.querySelector("#alarmSet");
+  let minutesSeconds = Number(setAlarmInput.value);
+  let stop = document.querySelector("#stop");
+  let set = document.querySelector("#set");
 
+  set.addEventListener("click", function () {
+    clearInterval(startCount);
+    if (setAlarmInput.value === "") {
+      alert("Please enter value");
+    }
+  });
+
+  stop.addEventListener("click", () => {
+    clearInterval(startCount);
+  });
+  let startCount = setInterval(function () {
+    let seconds = minutesSeconds % 60;
+    let minutes = Math.floor(minutesSeconds / 60);
+    if (seconds < 10) {
+      seconds = `0${seconds}`;
+    }
+    if (minutes < 10) {
+      minutes = `0${minutes}`;
+    }
+    if (minutesSeconds === 0) {
+      playAlarm();
+      clearInterval(startCount);
+    }
+
+    timeRemaining.innerHTML = `Time Remaining: ${minutes}:${seconds}`;
+    minutesSeconds--;
+  }, 1000);
+}
 // DO NOT EDIT BELOW HERE
```

Comment:
Overall looks good and appears to work correctly. Just a suggestion to make the code even better: consider breaking the code into smaller functions to improve readability and maintainability. For example, you could break out the logic for starting and stopping the timer into separate functions and improve readability and maintainability!